SYCL/ESIMD/radix_sort.cpp

```diff
@@ -575,7 +575,10 @@ void radix_sort(queue &q, unsigned *in, unsigned *out, unsigned size) {
     // next iteration takes out buffer as input
     //
     std::swap(in, out);
+    
   }
+  q.wait();
```

Comment:
The queue is created with in-order property. Will sycl host enforce the dependence so this change is not required? Adding wait after each inner iteration may impact performance. The test worked fine with the wait inserted only before the memcmp on host.

Reply:
Well, the wait is after the for loop and just before the method returns, so it should be effectively the same as inserting before memcpy on host. It was needed to ensure that variable being freed on the next line isn't being used by the kernels enqueued.